pkg/innerring/processors/alphabet/process_emit.go

```diff
@@ -9,15 +13,61 @@ func (np *Processor) processEmit() {
 	index := np.irList.Index()
 	if index < 0 {
 		np.log.Info("passive mode, ignore gas emission event")
+
 		return
 	} else if int(index) >= len(np.alphabetContracts) {
 		np.log.Debug("node is out of alphabet range, ignore gas emission event",
 			zap.Int32("index", index))
+
 		return
 	}
 
 	err := invoke.AlphabetEmit(np.morphClient, np.alphabetContracts[index])
 	if err != nil {
 		np.log.Warn("can't invoke alphabet emit method")
+
+		return
+	}
+
+	if np.storageEmission == 0 {
+		np.log.Info("storage node emission is off")
+
+		return
+	}
+
+	networkMap, err := invoke.NetmapSnapshot(np.morphClient, np.netmapContract)
+	if err != nil {
+		np.log.Warn("can't get netmap snapshot to emit gas to storage nodes",
+			zap.String("error", err.Error()))
+
+		return
+	}
+
+	ln := len(networkMap.Nodes)
+	if ln == 0 {
+		np.log.Debug("empty network map, do not emit gas")
+
+		return
+	}
+
+	gasPerNode := util.Fixed8(np.storageEmission / uint64(ln))
```

Comment:
If `np.storageEmission < ln` then `gasPerNode = 0`, is it correct behavior?

Reply:
That is a good point. The short answer: I don't know. 

In practice this should be very unlikely, because `np.storageEmission` is a `Fixed8` value. In my internal tests we can spend about 1~2 GAS in an epoch for storage nodes. So `gasPerNode` will be less than zero if there are >1000000 nodes in the network map. 

In this case spending 0 GAS is an predictable behavior. If there are more nodes in the network than storage emission value, then economic model should adapt: either increase storage emission or tranfer GAS to limited set of nodes. 

Both of these options have pros and cons and needs to be discussed. I think I'll create issue on that.

Reply:
https://github.com/nspcc-dev/neofs-node/issues/143